### PR TITLE
doc: network doc

### PR DESCRIPTION
**Pull request by driftluo**

### CODE REVIEW

File: network/src/errors.rs [hunk 1]
```diff
@@ -11,73 +11,73 @@ use std::fmt;
...
+    /// Handle panic or other error
     Protocol(ProtocolHandleErrorKind),
-    /// TODO(doc): @driftluo
+    /// Dail error
     Dail(DialerErrorKind),
-    /// TODO(doc): @driftluo
+    /// Listen error
     Listen(ListenErrorKind),
-    /// TODO(doc): @driftluo
+    /// Net shutdown or too many message block on
```
Comment (TheWaWaR) [inline network/src/errors.rs:45]:
```suggestion
    /// Net shutdown or too many messages block on
```

File: network/src/errors.rs [hunk 2]
```diff
@@ -11,73 +11,73 @@ use std::fmt;
...
 }
 
-/// TODO(doc): @driftluo
+/// Peer store error
 #[derive(Debug)]
 pub enum PeerStoreError {
     /// indicate the peer store is full
     EvictionFailed,
-    /// TODO(doc): @driftluo
+    /// file data is not json format
```
Comment (TheWaWaR) [inline network/src/errors.rs:54]: Please capitalize the first letter.

File: network/src/errors.rs [hunk 3]
```diff
@@ -11,73 +11,73 @@ use std::fmt;
...
     Serde(serde_json::Error),
 }
 
-/// TODO(doc): @driftluo
+/// Peer error
 #[derive(Debug, Eq, PartialEq)]
 pub enum PeerError {
-    /// TODO(doc): @driftluo
+    /// session already exist
```
Comment (TheWaWaR) [inline network/src/errors.rs:61]:
```suggestion
    /// session already exists
```

File: network/src/network.rs [hunk 4]
```diff
@@ -284,18 +273,20 @@ impl NetworkState {
         callback(&mut self.peer_store.lock())
     }
 
-    /// TODO(doc): @driftluo
+    /// Get self peer id
```
Comment (TheWaWaR) [inline network/src/network.rs:276]:
```suggestion
    /// Get peer id of local node
```

File: network/src/network.rs [hunk 5]
```diff
@@ -284,18 +273,20 @@ impl NetworkState {
...
     }
 
+    /// Use on test
+    #[allow(dead_code)]
     pub(crate) fn local_private_key(&self) -> &secio::SecioKeyPair {
         &self.local_private_key
     }
 
-    /// TODO(doc): @driftluo
+    /// Get self peer id base58 format string
```
Comment (TheWaWaR) [inline network/src/network.rs:287]:
```suggestion
    /// Get local node's peer id in base58 format string
```

File: network/src/network.rs [hunk 6]
```diff
@@ -311,7 +302,7 @@ impl NetworkState {
         self.peer_registry.read().connection_status()
     }
 
-    /// TODO(doc): @driftluo
+    /// Get self listen address list
```
Comment (TheWaWaR) [inline network/src/network.rs:305]:
```suggestion
    /// Get local node's listen address list
```

File: network/Cargo.toml [hunk 7]
```diff
@@ -4,7 +4,7 @@ version = "0.38.0-pre"
 license = "MIT"
 authors = ["Nervos Core Dev <dev@nervos.org>"]
 edition = "2018"
-description = "TODO(doc): @driftluo crate description"
+description = "ckb network impl"
```
Comment (TheWaWaR) [inline network/Cargo.toml:7]:
```suggestion
description = "ckb network implementation"
```

File: network/src/network.rs [hunk 8]
```diff
@@ -495,42 +486,42 @@ impl NetworkState {
...
+    pub(crate) fn add_observed_addrs(&self, iter: impl Iterator<Item = Multiaddr>) {
         let mut pending_observed_addrs = self.pending_observed_addrs.write();
         for addr in iter {
             trace!("pending observed addr: {:?}", addr,);
             pending_observed_addrs.insert(addr);
         }
     }
 
-    /// TODO(doc): @driftluo
+    /// network message processing controller, always true, if false, discard any received messages
```
Comment (TheWaWaR) [inline network/src/network.rs:498]: Whay `always true`?

File: network/src/network.rs [hunk 9]
```diff
@@ -1220,15 +1211,15 @@ impl NetworkController {
             .cloned()
     }
 
-    /// TODO(doc): @driftluo
+    /// Ban a ip
```
Comment (TheWaWaR) [inline network/src/network.rs:1214]:
```suggestion
    /// Ban an ip
```

File: network/src/network.rs [hunk 10]
```diff
@@ -1220,15 +1211,15 @@ impl NetworkController {
...
     pub fn ban(&self, address: IpNetwork, ban_until: u64, ban_reason: String) -> Result<(), Error> {
         self.network_state
             .peer_store
             .lock()
             .ban_network(address, ban_until, ban_reason)
     }
 
-    /// TODO(doc): @driftluo
+    /// Unban a ip
```
Comment (TheWaWaR) [inline network/src/network.rs:1222]:
```suggestion
    /// Unban an ip
```

File: network/src/network.rs [hunk 11]
```diff
@@ -1237,7 +1228,7 @@ impl NetworkController {
             .unban_network(address);
     }
 
-    /// TODO(doc): @driftluo
+    /// Return all connected peer info
```
Comment (TheWaWaR) [inline network/src/network.rs:1231]:
```suggestion
    /// Return all connected peer informations
```

File: network/src/network.rs [hunk 12]
```diff
@@ -1282,21 +1273,21 @@ impl NetworkController {
         }
     }
 
-    /// TODO(doc): @driftluo
+    /// Broadcast a message to all session
```
Comment (TheWaWaR) [inline network/src/network.rs:1276]:
```suggestion
    /// Broadcast a message to all sessions
```

File: network/src/network.rs [hunk 13]
```diff
@@ -1282,21 +1273,21 @@ impl NetworkController {
...
     pub fn broadcast(&self, proto_id: ProtocolId, data: Bytes) -> Result<(), SendErrorKind> {
         let session_ids = self.network_state.peer_registry.read().connected_peers();
         let target = TargetSession::Multi(session_ids);
         self.try_broadcast(false, target, proto_id, data)
     }
 
-    /// TODO(doc): @driftluo
+    /// Broadcast a message to all session through quick queue
```
Comment (TheWaWaR) [inline network/src/network.rs:1283]:
```suggestion
    /// Broadcast a message to all connected peers through quick queue
```

File: network/src/network.rs [hunk 14]
```diff
@@ -1282,21 +1273,21 @@ impl NetworkController {
...
     pub fn quick_broadcast(&self, proto_id: ProtocolId, data: Bytes) -> Result<(), SendErrorKind> {
         let session_ids = self.network_state.peer_registry.read().connected_peers();
         let target = TargetSession::Multi(session_ids);
         self.try_broadcast(true, target, proto_id, data)
     }
 
-    /// TODO(doc): @driftluo
+    /// Send message to one session
```
Comment (TheWaWaR) [inline network/src/network.rs:1290]:
```suggestion
    /// Send message to one connected peer
```

File: network/src/network.rs [hunk 15]
```diff
@@ -1307,22 +1298,22 @@ impl NetworkController {
...
     }
 
-    /// TODO(doc): @driftluo
+    /// network message processing controller, always true, if false, discard any received messages
     pub fn is_active(&self) -> bool {
         self.network_state.is_active()
     }
 
-    /// TODO(doc): @driftluo
+    /// Change active status, discard any received messages
```
Comment (TheWaWaR) [inline network/src/network.rs:1306]:
```suggestion
    /// Change active status, if set to false discard any received messages
```

File: network/src/network.rs [hunk 16]
```diff
@@ -1307,22 +1298,22 @@ impl NetworkController {
...
     pub fn set_active(&self, active: bool) {
         self.network_state.active.store(active, Ordering::Relaxed);
     }
 
-    /// TODO(doc): @driftluo
+    /// Return all connected session's protocols info
```
Comment (TheWaWaR) [inline network/src/network.rs:1311]:
```suggestion
    /// Return all connected peers' protocols info
```

File: network/src/network.rs [hunk 17]
```diff
@@ -1307,22 +1298,22 @@ impl NetworkController {
...
     pub fn protocols(&self) -> Vec<(ProtocolId, String, Vec<String>)> {
         self.network_state.protocols.read().clone()
     }
 
-    /// TODO(doc): @driftluo
+    /// Try ping all connected session
```
Comment (TheWaWaR) [inline network/src/network.rs:1316]:
```suggestion
    /// Try ping all connected peers
```

File: network/src/peer_registry.rs [hunk 18]
```diff
@@ -26,18 +26,18 @@ pub struct PeerRegistry {
     feeler_peers: HashSet<PeerId>,
 }
 
-/// TODO(doc): @driftluo
+/// Global network connection status
 #[derive(Clone, Copy, Debug)]
 pub struct ConnectionStatus {
-    /// TODO(doc): @driftluo
+    /// Total connected session number
```
Comment (TheWaWaR) [inline network/src/peer_registry.rs:32]:
```suggestion
    /// Total session number
```

File: network/src/peer_registry.rs [hunk 19]
```diff
@@ -232,12 +232,12 @@ impl PeerRegistry {
             .and_then(|session_id| self.peers.remove(&session_id))
     }
 
-    /// TODO(doc): @driftluo
+    /// Get all connected peer info
```
Comment (TheWaWaR) [inline network/src/peer_registry.rs:235]:
```suggestion
    /// Get all connected peers' information
```

File: network/src/peer_registry.rs [hunk 20]
```diff
@@ -232,12 +232,12 @@ impl PeerRegistry {
...
     pub fn peers(&self) -> &HashMap<SessionId, Peer> {
         &self.peers
     }
 
-    /// TODO(doc): @driftluo
+    /// Get all connected session id
```
Comment (TheWaWaR) [inline network/src/peer_registry.rs:240]:
```suggestion
    /// Get all sessions' id
```

File: network/src/peer_store/addr_manager.rs [hunk 21]
```diff
@@ -13,7 +13,7 @@ pub struct AddrManager {
 }
 
 impl AddrManager {
-    /// TODO(doc): @driftluo
+    /// add
```
Comment (TheWaWaR) [inline network/src/peer_store/addr_manager.rs:16]:
```suggestion
    /// Add an address information to address manager
```

File: network/src/network.rs (hunk 8 above)
Comment (driftluo) [reply network/src/network.rs:498]: It is always true unless the user modifies it through rpc

File: network/src/peer_store/addr_manager.rs [hunk 22]
```diff
@@ -65,17 +65,17 @@ impl AddrManager {
         addr_infos
     }
 
-    /// TODO(doc): @driftluo
+    /// count
```
Comment (TheWaWaR) [inline network/src/peer_store/addr_manager.rs:68]:
```suggestion
    /// The count of address in address manager
```

File: network/src/peer_store/addr_manager.rs [hunk 23]
```diff
@@ -65,17 +65,17 @@ impl AddrManager {
...
     }
 
-    /// TODO(doc): @driftluo
+    /// addresses iter
     pub fn addrs_iter(&self) -> impl Iterator<Item = &AddrInfo> {
         self.id_to_info.values()
     }
 
-    /// TODO(doc): @driftluo
+    /// remove address
```
Comment (TheWaWaR) [inline network/src/peer_store/addr_manager.rs:78]:
```suggestion
    /// Remove an address by its ip and port
```

File: network/src/peer_store/addr_manager.rs [hunk 24]
```diff
@@ -88,14 +88,14 @@ impl AddrManager {
         }
     }
 
-    /// TODO(doc): @driftluo
+    /// get info from ip port
```
Comment (TheWaWaR) [inline network/src/peer_store/addr_manager.rs:91]:
```suggestion
    /// Get an address information by ip and port
```

File: network/src/peer_store/ban_list.rs [hunk 25]
```diff
@@ -57,13 +57,13 @@ impl BanList {
         })
     }
 
-    /// TODO(doc): @driftluo
+    /// whether ip banned
```
Comment (TheWaWaR) [inline network/src/peer_store/ban_list.rs:60]:
```suggestion
    /// Whether the ip is banned
```

File: network/src/peer_store/ban_list.rs [hunk 26]
```diff
@@ -57,13 +57,13 @@ impl BanList {
...
     pub fn is_ip_banned(&self, ip: &IpAddr) -> bool {
         let now_ms = unix_time_as_millis();
         self.is_ip_banned_until(ip.to_owned(), now_ms)
     }
 
-    /// TODO(doc): @driftluo
+    /// whether address banned
```
Comment (TheWaWaR) [inline network/src/peer_store/ban_list.rs:66]:
```suggestion
    /// Whether the address is banned
```

File: network/src/peer_store/ban_list.rs [hunk 27]
```diff
@@ -72,7 +72,7 @@ impl BanList {
         false
     }
 
-    /// TODO(doc): @driftluo
+    /// get banned list
```
Comment (TheWaWaR) [inline network/src/peer_store/ban_list.rs:75]:
```suggestion
    /// Get banned address list
```

File: network/src/peer_store/mod.rs [hunk 28]
```diff
@@ -45,28 +45,28 @@ impl Default for PeerScoreConfig {
...
 
-/// TODO(doc): @driftluo
+/// report result
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
 pub enum ReportResult {
-    /// TODO(doc): @driftluo
+    /// ok
     Ok,
-    /// TODO(doc): @driftluo
+    /// ban
```
Comment (TheWaWaR) [inline network/src/peer_store/mod.rs:59]:
```suggestion
    /// The peer is banned
```

File: network/src/peer_store/mod.rs [hunk 29]
```diff
@@ -45,28 +45,28 @@ impl Default for PeerScoreConfig {
 /// Peer Status
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
 pub enum Status {
-    /// TODO(doc): @driftluo
+    /// connected
     Connected,
-    /// TODO(doc): @driftluo
+    /// disconnected
```
Comment (TheWaWaR) [inline network/src/peer_store/mod.rs:50]:
```suggestion
    /// The peer is disconnected
```

File: network/src/peer_store/peer_store_db.rs [hunk 30]
```diff
@@ -16,15 +16,15 @@ const DEFAULT_ADDR_MANAGER_DB: &str = "addr_manager.db";
...
+    /// load address list from disk
     pub fn load<R: Read>(r: R) -> Result<Self, Error> {
         let addrs: Vec<AddrInfo> = serde_json::from_reader(r).map_err(PeerStoreError::Serde)?;
         let mut addr_manager = AddrManager::default();
         addrs.into_iter().for_each(|addr| addr_manager.add(addr));
         Ok(addr_manager)
     }
 
-    /// TODO(doc): @driftluo
+    /// dump address list to disck
```
Comment (TheWaWaR) [inline network/src/peer_store/peer_store_db.rs:27]:
```suggestion
    /// Dump address list to disk
```

File: network/src/network.rs (hunk 8 above)
Comment (TheWaWaR) [reply network/src/network.rs:498]: Better say "default is true".

File: network/src/peer_store/peer_store_impl.rs [hunk 31]
```diff
@@ -203,22 +203,22 @@ impl PeerStore {
         Ok(())
     }
 
-    /// TODO(doc): @driftluo
+    /// whether address banned
```
Comment (TheWaWaR) [inline network/src/peer_store/peer_store_impl.rs:206]:
```suggestion
    /// whether the address is banned
```

File: network/src/errors.rs [hunk 32]
```diff
@@ -11,73 +11,73 @@ use std::fmt;
...
+    /// Handle panic or other error
     Protocol(ProtocolHandleErrorKind),
-    /// TODO(doc): @driftluo
+    /// Dail error
     Dail(DialerErrorKind),
-    /// TODO(doc): @driftluo
+    /// Listen error
     Listen(ListenErrorKind),
-    /// TODO(doc): @driftluo
+    /// Net shutdown or too many message blocked on
```
Comment (chuijiaolianying) [inline network/src/errors.rs:45]:
```suggestion
    /// Net shutdown or too many messages blocked on
```

File: network/src/lib.rs [hunk 33]
```diff
@@ -1,4 +1,11 @@
-//! TODO(doc): @driftluo
+//! ckb network module
+//!
+//! This module is based on the Tentacle library, once again abstracting the context that protocols can use,
```
Comment (chuijiaolianying) [inline network/src/lib.rs:3]:
```suggestion
//! This module is based on the Tentacle library, once again abstract the context that protocols can use,
```

File: network/src/network.rs [hunk 34]
```diff
@@ -495,42 +486,42 @@ impl NetworkState {
...
 
-/// TODO(doc): @driftluo
+/// Used to handle global events of tentacle, such as session open/close
 pub struct EventHandler<T> {
     pub(crate) network_state: Arc<NetworkState>,
     pub(crate) exit_handler: T,
 }
 
-/// TODO(doc): @driftluo
+/// Exit trait use to notify all other module to exit
```
Comment (chuijiaolianying) [inline network/src/network.rs:510]:
```suggestion
/// Exit trait used to notify all other module to exit
```

File: network/src/peer_registry.rs [hunk 35]
```diff
@@ -187,27 +187,27 @@ impl PeerRegistry {
...
     }
 
-    /// TODO(doc): @driftluo
+    /// Add feeler dail task
     pub fn add_feeler(&mut self, peer_id: PeerId) {
         self.feeler_peers.insert(peer_id);
     }
 
-    /// TODO(doc): @driftluo
+    /// Remove feeler dail task on session disconnect or fails
```
Comment (chuijiaolianying) [inline network/src/peer_registry.rs:195]:
```suggestion
    /// Remove feeler dail task on session disconnects or fails
```

File: network/src/peer_store/addr_manager.rs [hunk 36]
```diff
@@ -65,17 +65,17 @@ impl AddrManager {
...
     }
 
-    /// TODO(doc): @driftluo
+    /// The count of address in address manager
     pub fn count(&self) -> usize {
         self.addr_to_id.len()
     }
 
-    /// TODO(doc): @driftluo
+    /// Addresses iter
```
Comment (chuijiaolianying) [inline network/src/peer_store/addr_manager.rs:73]:
```suggestion
    /// Addresses iterator
```

File: sync/src/lib.rs [hunk 37]
```diff
@@ -21,27 +21,18 @@ pub use crate::synchronizer::Synchronizer;
 pub use crate::types::SyncShared;
 use std::time::Duration;
 
-/// TODO(doc): @driftluo
+/// Default max get header response len, if it is greater than this value, the message will be ignored
```
Comment (chuijiaolianying) [inline sync/src/lib.rs:24]:
```suggestion
/// Default max get header response length, if it is greater than this value, the message will be ignored
```

File: sync/src/lib.rs [hunk 38]
```diff
@@ -56,54 +47,49 @@ pub(crate) const LOW_INDEX: usize = TIME_TRACE_SIZE * 9 / 10;
...
+/// Interval between repeated inquiry transactions
 pub const RETRY_ASK_TX_TIMEOUT_INCREASE: Duration = Duration::from_secs(30);
 
-/// TODO(doc): @driftluo
+/// Default ban time for message
 // ban time
 // 5 minutes
 pub const BAD_MESSAGE_BAN_TIME: Duration = Duration::from_secs(5 * 60);
-/// TODO(doc): @driftluo
+/// Default ban tim for sync useless
```
Comment (chuijiaolianying) [inline sync/src/lib.rs:93]:
```suggestion
/// Default ban time for sync useless
```

File: sync/src/status.rs [hunk 39]
```diff
@@ -159,12 +159,12 @@ impl Status {
         }
     }
 
-    /// TODO(doc): @driftluo
+    /// Whether should output a warn log
```
Comment (chuijiaolianying) [inline sync/src/status.rs:162]:
```suggestion
    /// Whether should output a warning log
```

File: sync/src/types/mod.rs [hunk 40]
```diff
@@ -1442,14 +1442,14 @@ impl SyncShared {
             .or_else(|| self.store().get_block_header(hash))
     }
 
-    /// TODO(doc): @driftluo
+    /// Check whether block's parent has been insert to chain store
```
Comment (chuijiaolianying) [inline sync/src/types/mod.rs:1445]:
```suggestion
    /// Check whether block's parent has been inserted to chain store
```

File: util/network-alert/src/notifier.rs [hunk 41]
```diff
@@ -68,7 +68,7 @@ impl Notifier {
         true
     }
 
-    /// TODO(doc): @driftluo
+    /// Add a alert
```
Comment (chuijiaolianying) [inline util/network-alert/src/notifier.rs:71]:
```suggestion
    /// Add an alert
```

**Review by chuijiaolianying — DISMISSED** [review]
LGTM

**Review by yangby-cryptape — COMMENTED** [review]
- Still not all sentences are capitalized.
- In my humble opinion, using simple abbreviations in comments is not suitable.

  For example: "Init", "sync", "ext" and so on.

  Yes, all of us can understand these abbreviations, but I think these comments are not only for us.
  Full and clear words are more formal, and more friendly for those beginners who may not understand those abbreviations.

  p.s. I checked [The Rust Standard Library Documentation](https://doc.rust-lang.org/std/index.html), they use full words in comments even the function names used abbreviations.

- Too brief, from the perspective of readers. For example:
  - If a function named `new` and the only comment is `Init`.
  - If a module is `verifier.rs` and the only comment for the module is `verify module`.
  - Some items in the enumerated type `SupportProtocols`:
    - The comment of `DisconnectMessage` is "Disconnect message", OK, em, ..., I can understand,
    - The comment of `Feeler` is "Feeler", em, ..., :scream:, what's "Feeler"? Did I miss something? 
    - Well, I find a file named `feeler.rs`, :thinking: let me open it!
    - Oh, no, :sob: the only comment is "Feeler", too! Fxxk!
    - Let me [`Just Fxxking Google It`](http://www.justfuckinggoogleit.com/), what is "CKB Feeler": "Did you mean: CURB feeler or KERB feeler". :skull:
    - CKB (documentation) is sucks. :-1:


  Readers can only get one message from these comments: their boss told them that the comments rate is related to KPI. :zipper_mouth_face:

  I don't think all comments should be lengthy, but some comments are really too brief.

  Sometimes, maybe leave few references is better.

###  P.S.

As an example, I think the comment of `Feeler` should be "Feeler Connection is intended to test a peer is connectable or not.". And we also could add a link to [CKB RFCs](https://github.com/nervosnetwork/rfcs/blob/v2020.01.15/rfcs/0007-scoring-system-and-network-security/0007-scoring-system-and-network-security.md#feeler-connection) or other more detailed articles or papers, such as [Eclipse Attacks on Bitcoin's Peer-to-Peer Network](https://cryptographylab.bitbucket.io/slides/Eclipse%20Attacks%20on%20Bitcoin%27s%20Peer-to-Peer%20Network.pdf).

**Comment by driftluo** [comment]
> If a function named new and the only comment is Init.
Please provide a modification plan, I can no longer comment correctly on the new function
> If a module is verifier.rs and the only comment for the module is verify module.
I will add some warnings about misbehaving
> Some items in the enumerated type SupportProtocols
I will add a description of the role of the protocol

**Comment by yangby-cryptape** [comment]
> have nothing to say about the function or structure

Yep, you aren't alone.  :handshake::handshake::handshake: **Sometimes I have the same feeling too!**

I know these items very clear, I don't think they need more explanations.
In Chinese, there is a verse to describe that: "不识庐山真面目，只缘身在此山中". 

But for beginners, sometime there are really difficult to understand without more explanations.
And looks really don't formal.

p.s. **I don't mean that all comments should be lengthy.**

> a modification plan

Just an example in my opinion, maybe other reviewers can give us better solutions.

```rust
impl Relayer {
    /// Creates a new `Relayer`. (or "Initialize a new `Relayer`" is ok, too)
    ///
    /// If in CKB there is no introduction of `Relayer`, here could be a simple 
    /// introduction for `Relayer`, otherwise a link to the introduction is enough.
    ///
    /// `rate_limiter` is not in this function arguments, we could point out which
    /// default value is used here. Move the followed comments of the default 
    /// `rate_limiter` to here (there are already some comments for that).
    pub fn new(..) -> Self { .. }
}
```

Refers to the follow items in [The Rust Standard Library Documentation](https://doc.rust-lang.org/std/index.html):
- [String::new(..)](https://doc.rust-lang.org/std/string/struct.String.html#method.new)
- [HashMap::new(..)](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.new)

**Comment by doitian** [comment]
> The short comment is just because I have nothing to say about the function or structure
> 
> > If a function named new and the only comment is Init.
> 
> Please provide a modification plan, I can no longer comment correctly on the new function
> 
> > If a module is verifier.rs and the only comment for the module is verify module.
> 
> I will add some warnings about misbehaving
> 
> > Some items in the enumerated type SupportProtocols
> 
> I will add a description of the role of the protocol

When there’s nothing to say, show some examples.

**Comment by driftluo** [comment]
> When there’s nothing to say, show some examples.

Such as peer store new, peer registry new

**Comment by chuijiaolianying** [comment]
> > When there’s nothing to say, show some examples.
> 
> Such as peer store new, peer registry new

I think ian means you could add some examples of the code, not ask for examples of 'nothing to say', 😂

**Comment by driftluo** [comment]
> > > When there’s nothing to say, show some examples.
> > 
> > 
> > Such as peer store new, peer registry new
> 
> I think ian means you could add some examples of the code, not ask for examples of 'nothing to say', joy

The example cannot be added either. . . Because there is only one way to initialize

**Review by chuijiaolianying — APPROVED** [review]

**Review by TheWaWaR — APPROVED** [review]

**Review by doitian — APPROVED** [review]
Let's merge it first and improve the words later.

**Comment by doitian** [comment]
bors r=TheWaWaR,chuijiaolianying

**Comment by yangby-cryptape** [comment]
bors retry

**Pull request merged**